oracle/src/main.rs

```diff
@@ -0,0 +1,37 @@
+use clap::Parser;
+use cli_ext::OracleExt;
+use exex::ExEx;
+use network::{proto::OracleProtoHandler, Network};
+use oracle::Oracle;
+use reth_network::{protocol::IntoRlpxSubProtocol, NetworkProtocols};
+use reth_node_ethereum::EthereumNode;
+
+mod cli_ext;
+mod exex;
+mod network;
+mod oracle;
+
+const ORACLE_EXEX_ID: &str = "exex-oracle";
+
+fn main() -> eyre::Result<()> {
+    reth::cli::Cli::<OracleExt>::parse().run(|builder, args| async move {
+        let tcp_port = args.tcp_port;
+        let udp_port = args.udp_port;
+
+        let handle = builder
+            .node(EthereumNode::default())
+            .install_exex(ORACLE_EXEX_ID, move |ctx| async move {
+                let subproto = OracleProtoHandler::new();
```

Comment:
here we'd need shared state between the protocolhandler and the ext, for example the ex monitors state changes and sends data through that shared state (channels)